app/models/compute_resources/foreman/model/openstack.rb

```diff
@@ -139,6 +139,11 @@ def create_vm(args = {})
       raise message
     end
 
+    def vm_ready(vm)
+      vm.wait_for { self.ready? || self.failed? }
+      raise "Failed to deploy vm %{name}, fault: %{e}" % { :name => vm.name, :e => vm.fault } if vm.failed?
```

Comment:
In addition, why not use vm.fault['message'] to display only the error message and not the http status code and timestamp?